### PR TITLE
remove duplicate code in citus_dist_stat_activity

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**

### CODE REVIEW

File: src/backend/distributed/transaction/citus_dist_stat_activity.c [hunk 1]
```diff
@@ -253,6 +253,8 @@ static TimestampTz ParseTimestampTzFieldFromHeapTuple(HeapTuple tuple, TupleDesc
 													  int colIndex);
 static TransactionId ParseXIDFieldFromHeapTuple(HeapTuple tuple, TupleDesc tupdesc, int
 												colIndex);
+static void ReplaceInitiatorNodeIdentifier(int initiator_node_identifier,
```
Comment (onderkalaci) [inline src/backend/distributed/transaction/citus_dist_stat_activity.c]: Super minor coding convention notes:

We usually prefer to declare & define the function right after the first function that uses them. In this case, it'd be good to move the declaration right after `ParseCitusDistStat `. As far as I can see, the definition is already on the correct place.

File: src/backend/distributed/transaction/citus_dist_stat_activity.c [hunk 2]
```diff
@@ -487,55 +489,11 @@ ParseCitusDistStat(PGresult *result, int64 rowIndex)
...
-	{
-		citusDistStat->master_query_host_name = cstring_to_text(coordinator_host_name);
-		citusDistStat->master_query_host_port = 0;
-	}
-	else
-	{
-		citusDistStat->master_query_host_name = NULL;
-		citusDistStat->master_query_host_port = 0;
-	}
+	ReplaceInitiatorNodeIdentifier(initiator_node_identifier, citusDistStat);
```
Comment (onderkalaci) [inline src/backend/distributed/transaction/citus_dist_stat_activity.c]: Again super minor convention note, feel free to ignore: We generally try to have a single empty line between code-blocks that are logically different than others.

For example, when you use  `ReplaceInitiatorNodeIdentifier ` is feels a little bit different than `PQgetisnull ` / `ParseIntFieldFromHeapTuple `, so might be good to have a single line for readability purpose.

Try it out, if that doesn't feel similar for you, skip doing it.

File: src/backend/distributed/transaction/citus_dist_stat_activity.c (hunk 1 above)
Comment (SaitTalhaNisanci) [reply src/backend/distributed/transaction/citus_dist_stat_activity.c]: putting definition right after the usage makes it easy to follow the flow of the code, but what benefit does having declaration in the same order?

(I have updated the declaration too)

Comment (onderkalaci) [reply src/backend/distributed/transaction/citus_dist_stat_activity.c]: I think you can trace the related functions a bit easier

**Pull request merged**